tests/routerTest.php

```diff
@@ -70,6 +72,22 @@ public function testCloudStorageStreamWrapperNotRegisteredByDefault(): void
         $this->assertEquals($wrappers, stream_get_wrappers());
         $this->assertNotContains('gs', stream_get_wrappers());
     }
+
+    public function testDeclarativeOverNonDeclarative(): void
+    {
+        FunctionsFramework::http('helloHttp', function (ServerRequestInterface $request) {
+            return 'Hello World!';
+        });
+
+        // index.php also has a non-declaration function named 'helloHttp'.
+        putenv('FUNCTION_SOURCE=' . __DIR__ . '/../examples/hello/index.php');
+        putenv('FUNCTION_TARGET=helloHttp');
+        putenv('FUNCTION_SIGNATURE_TYPE=cloudevent'); // ignored due to declarative signature
+        require 'router.php';
+
+        // Expect the declarative function to be called.
+        $this->expectOutputString('Hello World!');
```

Comment:
This is a super-nit, but to be absolutely sure the function in `hello/index` isn't being called, I would suggest using `"Hello World from a declarative function"` or some other less generic string.